ruby/lib/ci/queue/configuration.rb

```diff
@@ -1,21 +1,25 @@
+require 'ci/queue/null_flaky_supplier'
+require 'ci/queue/file_flaky_supplier'
+
 module CI
   module Queue
     class Configuration
-      attr_accessor :timeout, :build_id, :worker_id, :max_requeues, :requeue_tolerance, :namespace, :seed, :failing_test
+      attr_accessor :timeout, :build_id, :worker_id, :max_requeues, :requeue_tolerance, :namespace, :seed, :failing_test, :flaky_test_supplier
 
       class << self
         def from_env(env)
           new(
             build_id: env['CIRCLE_BUILD_URL'] || env['BUILDKITE_BUILD_ID'] || env['TRAVIS_BUILD_ID'],
             worker_id: env['CIRCLE_NODE_INDEX'] || env['BUILDKITE_PARALLEL_JOB'],
             seed: env['CIRCLE_SHA1'] || env['BUILDKITE_COMMIT'] || env['TRAVIS_COMMIT'],
+            flaky_test_supplier: (env['FLAKY_TEST_SUPPLIER_FILE'] && CI::Queue::FileFlakySupplier.new(env['FLAKY_TEST_SUPPLIER_FILE'])) || CI::Queue::NullFlakySupplier,
```

Comment:
I think it would be better to namespace that variable. Something like `CI_QUEUE_FLAKY_TESTS`.